spring-cloud-gateway-mvc/src/main/java/org/springframework/cloud/gateway/mvc/ProxyExchange.java

```diff
@@ -251,11 +253,23 @@ public ProxyExchange<T> uri(String uri) {
 		}
 	}
 
+	/**
+	 * Returns the request path without any query parameters.
+	 * @return the URI path
+	 * @see ProxyExchange#path(String)
+	 * @see ProxyExchange#queryParams
+	 */
 	public String path() {
-		return (String) this.webRequest.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE,
-				WebRequest.SCOPE_REQUEST);
+		return exchange.getRequest().getPath().pathWithinApplication().value();
```

Comment:
There's no exchange in spring mvc, this doesn't even compile

Reply:
There was at the time I opened this MR.

Reply:
Server Web exchange is from spring webflux